python/pynessie/cli.py

```diff
@@ -236,6 +245,8 @@ def log(
         filtering_args["before"] = until
     if end:
         filtering_args["end"] = end
+    if cel_expr:
```

Comment:
I thought we were going to remove all query params but the expression?

Reply:
I think it's probably easier to still send the same query params to the server as before and let the server deal with the translation of those parameters to CEL rather than having to translate this on the client-side

Reply:
I don't agree. I think there should be one way of doing things w/ the REST API. An implementer shouldn't guess what will happen to their query if they send query params and an expression

Reply:
I believe we still have to support the same query params on the REST API (`authors` / `committers`, `before`, `after`, ...) as otherwise we're breaking the API for older clients. Even though we're not on 1.0 yet, it feels wrong to break this.

Also having the `CELTranslator` on the server makes it much easier imo than moving this to the client-side. 

If a user ends up using the `query_expression` param with other parameters, then we'll immediately let them know by throwing an error.

Overall I believe we don't want to make people have to use & learn CEL on the client-side if they don't want to. Users should still be able to say `nessie log --author X` rather than having to learn CEL and being forced to use `nessie log --expression commit.author = 'X'`. Using `--expression` / `query_expression` should be an additional option for users that want to use **advanced filtering** and are willing to look at CEL.

Reply:
>Overall I believe we don't want to make people have to use & learn CEL on the client-side if they don't want to. Users should still be able to say nessie log --author X rather than having to learn CEL and being forced to use nessie log --expression commit.author = 'X'. Using --expression / query_expression should be an additional option for users that want to use advanced filtering and are willing to look at CEL.

I agree. I am not suggesting we make users learn expressions just to do simple queries on the CLI. I am suggesting doing the translation to a CEL expression on the client side.

>If a user ends up using the query_expression param with other parameters, then we'll immediately let them know by throwing an error.

This is a runtime error rather than something they understand while reading docs. I think we should make the REST API as clear and well defined as possible. The ambiguity comes while reading the API docs rather than at implementation time. It is not clear to me why a design would support two ways of achieving the same thing via the REST API.

>  Even though we're not on 1.0 yet, it feels wrong to break this.

I disagree, the query params have been available for ~1 week in a pre-GA release. No one will notice/complain that they disappear in another week.

Reply:
@laurentgo / @snazy pinging you here so that you guys have some context as I wanted to bring this topic up in our standup so that we can decide on the optimal way going forward

Reply:
We had a good discussion in the standup and the summary of the discussion is:
* we don't care much about backwards compatility at this point. Also Iceberg and won't notice these breaking changes
* REST API will only support `query_expression` and get rid of parameters, such as `authors` / `committers` / `before` / `after`
* Python and Java client will send only the `query_expression` parameter
* Python CLI can still have `--author` / `--committer` / `--before`/ `--after` flags. Those will be **shortcuts** only and will be translated directly to a CEL expression in the CLI, thus only sending a `query_expression` to the server. Those **shortcus** should probably not be used in combination with the `--query-expression` flag.

I'll go ahead and implement those changes tomorrow.

Reply:
fixed as described in the last comment